src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -223,18 +223,23 @@ public void AllowReadStreamBuffering_SetTrueThenGet_ExpectTrue(Uri remoteServer)
             Assert.True(request.AllowReadStreamBuffering);
         }
 
-        [Theory, MemberData(nameof(EchoServers))]
-        public async Task ContentLength_Get_ExpectSameAsGetResponseStream(Uri remoteServer)
+        [Fact]
+        public async Task ContentLength_Get_ExpectSameAsGetResponseStream()
         {
-            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
-            using (WebResponse response = await request.GetResponseAsync())
-            using (Stream myStream = response.GetResponseStream())
-            using (StreamReader sr = new StreamReader(myStream))
+            await LoopbackServer.CreateServerAsync(async (server, uri) =>
             {
-                string strContent = sr.ReadToEnd();
-                long length = response.ContentLength;
-                Assert.Equal(strContent.Length, length);
-            }
+                HttpWebRequest request = WebRequest.CreateHttp(uri);
+                Task<WebResponse> getResponse = request.GetResponseAsync();
+                await server.HandleRequestAsync();
+                using (WebResponse response = await getResponse)
+                using (Stream myStream = response.GetResponseStream())
+                using (var sr = new StreamReader(myStream))
+                {
+                    string strContent = sr.ReadToEnd();
+                    long length = response.ContentLength;
+                    Assert.Equal(strContent.Length, length);
+                }
+            });
```

Comment:
Nit: we generally use CreateClientAndServerAsync, e.g.
```C#
await LoopbackServer.CreateServerAsync(async uri =>
{
    HttpWebRequest request = WebRequest.CreateHttp(uri);
    using (WebResponse response = await request.GetResponseAsync())
    ... // etc
},
server => server.HandleRequestAsync());
```

Reply:
Your example shows `CreateServerAsync` but you recommended to use `CreateClientAndServerAsync`. Am I missing something here?

Reply:
That was a typo... the signature I used was for CreateClientAndServerAsync, I just copy-pasted the wrong name from your code.